packages/windows_application_windows.go

```diff
@@ -0,0 +1,122 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package packages
+
+import (
+	//"golang.org/x/sys/windows"
+	//"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	//"time"
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/osconfig/clog"
+	"golang.org/x/sys/windows/registry"
+)
+
+func getStringValueReturnEmptyIfError(k *registry.Key, value string) string {
+	value, _, err := k.GetStringValue(value)
+	if err != nil {
+		return ""
+	}
+	return value
+}
+
+func parseDate(dateString string) time.Time {
+	var t time.Time
```

Comment:
time.Time isn't a reference type -> can't be nil. In this case error can be considered that it's not valid date, but we discussed on chat/gvc/somewhere else that it should just be "empty", so this function might as well just return the empty value if date isn't valid.

Reply:
In this case instantiate during the return so you don't create the unused variable. Another option would be to use a pointer so it can be nil (would be the preference if you also returned an error), but in this case that may cause additional logic in the caller so just returning time.Time{} is fine.